analyzers/tests/SonarAnalyzer.Test/TestCases/UnusedPrivateMember.CSharp9.cs

```diff
@@ -168,3 +169,121 @@ public class ClassWithPrintMembers
         }
     }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/9379
+namespace Repro_9379
+{
+    public static class Program
+    {
+        public static void Method()
+        {
+            var instance = CreateInstance<ClassInstantiatedThroughReflection>();
+
+            A classViaReflection = new();
+            InitValue(classViaReflection); 
+        }
+
+        public static T CreateInstance<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>() =>
+            (T)Activator.CreateInstance(typeof(T), 42);
```

Comment:
According [to the docs](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.dynamicallyaccessedmembersattribute?view=net-8.0#remarks), the attribute also works for `string` and `Type` parameters:
```cs
public static object CreateInstance([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type type) => // ..

CreateInstance(typeof(SomePrivateClass));
```

You may want to add an FP UT.